.github/workflows/continous-integration.yml

```diff
@@ -47,72 +53,27 @@ jobs:
           token: ${{ secrets.GITHUB_TOKEN }}
           filters: .github/change_filters.yml
 
-  documentation:
-    name: Test Documentation
+
+  wait_for_docs_tests:
+    # Looks for doc test workflows and waits for it to complete successfully
+    # Runs on tag pushes and pushes to main exclusively, as it is a dependency of release jobs
+    name: Wait for docs tests
     runs-on: ubuntu-latest
-    needs: [changes]
 
     steps:
-    - name: Checkout git repository 🕝
-      if: needs.changes.outputs.docs == 'true'
-      uses: actions/checkout@v2
+      - name: Checkout git repository 🕝
+        uses: actions/checkout@v2
 
-    - name: Set up Python 3.7 🐍
-      if: needs.changes.outputs.docs == 'true'
-      uses: actions/setup-python@v1
-      with:
-        python-version: 3.7
-
-    - name: Set up Node 12.x 🦙
-      if: needs.changes.outputs.docs == 'true'
-      uses: actions/setup-node@v1
-      with:
-        node-version: '12.x'
-
-    - name: Read Poetry Version 🔢
-      if: needs.changes.outputs.docs == 'true'
-      run: |
-        echo "POETRY_VERSION=$(scripts/poetry-version.sh)" >> $GITHUB_ENV
-      shell: bash
-
-    - name: Install poetry 🦄
-      if: needs.changes.outputs.docs == 'true'
-      uses: Gr1N/setup-poetry@v4
-      with:
-        poetry-version: ${{ env.POETRY_VERSION }}
-
-    - name: Load Poetry Cached Libraries ⬇
-      if: needs.changes.outputs.docs == 'true'
-      uses: actions/cache@v2
-      with:
-        path: ~/.cache/pypoetry/virtualenvs
-        key: ${{ runner.os }}-poetry-${{ env.POETRY_VERSION }}-3.7-${{ hashFiles('**/poetry.lock') }}-${{ secrets.POETRY_CACHE_VERSION }}
-        restore-keys: ${{ runner.os }}-poetry-3.7
-
-    - name: Load Yarn Cached Packages ⬇
-      if: needs.changes.outputs.docs == 'true'
-      uses: actions/cache@v2
-      with:
-        path: docs/node_modules
-        key: ${{ runner.os }}-yarn-12.x-${{ hashFiles('docs/yarn.lock') }}
-        restore-keys: ${{ runner.os }}-yarn-12.x
-
-    - name: Install Dependencies 📦
-      if: needs.changes.outputs.docs == 'true'
-      run: |
-        sudo apt-get -y install libpq-dev
-        make install-full install-docs
-
-    - name: Run Swagger 🕵️‍♀️
-      if: needs.changes.outputs.docs == 'true'
-      run: |
-        npm install -g swagger-cli
-        swagger-cli validate docs/static/spec/action-server.yml
-        swagger-cli validate docs/static/spec/rasa.yml
+      - name: Wait for doc tests
```

Comment:
It looks like it doesn't work

```
Retrieving check runs named Test documentation on RasaHQ/rasa@8aae074786c6058a0f649a9291dd6d788b643d96...
Retrieved 0 check runs named Test documentation
No completed checks named Test documentation, waiting for 60 seconds...
Retrieving check runs named Test documentation on RasaHQ/rasa@8aae074786c6058a0f649a9291dd6d788b643d96...
Retrieved 0 check runs named Test documentation
No completed checks named Test documentation, waiting for 60 seconds...
```

Reply:
This might because of bad logic in the workflow. Let me refactor it.

Reply:
Fixed the time out problem with a correct commit SHA ref, which is used to pull check runs.

`${{ github.event.pull_request.head.sha || github.sha }}` -> `${{ github.head_ref }}`

Reply:
> The head_ref or source branch of the pull request in a workflow run. This property is only available when the event that triggers a workflow run is a pull_request.

It doesn't work for non PR events

Reply:
Aw. There are also trigger events like tag push or commit push. How about `${{ github.head_ref || github.sha }}`?

Reply:
`${{ github.head_ref || github.sha }}` should be ok

Reply:
Committed!